runners/direct-java/src/test/java/org/apache/beam/runners/direct/UnboundedReadEvaluatorFactoryTest.java

```diff
@@ -458,6 +458,9 @@ private TestUnboundedSource(Coder<T> coder, boolean throwOnClose, List<T> elems)
     @Override
     public UnboundedSource.UnboundedReader<T> createReader(
         PipelineOptions options, @Nullable TestCheckpointMark checkpointMark) {
+      checkState(
+          checkpointMark == null || checkpointMark.decoded,
+          "Cannot resume from a checkpoint that has not been decoded");
```

Comment:
This is a `checkState` - there's either no checkpoint or there's a decoded checkpoint; if we get a non-null non-decoded checkpoint, then we're resuming from a state we don't expect and the test should fail.